Classes/Creator/Tca/Table/TcaTableCreator.php

```diff
@@ -108,17 +119,65 @@ private function addNewTcaTypeItems(array $existingTcaTypes, TableInformation $t
 
         /** @var String_ $showItems */
         $showItems = $existingTcaTypes[0]->value->items[0]->value;
-        $existingColumnNames = GeneralUtility::trimExplode(',', $showItems->value, true);
+        $showItemsString = $showItems->value;
 
+        // Normalize: remove all whitespace and check if fields already exist anywhere
+        $normalized = str_replace(["\n", "\r", "\t", ' '], '', $showItemsString);
+        $existingFieldNames = explode(',', $normalized);
+
+        // Collect fields to add
+        $newFields = [];
         foreach ($tableInformation->getColumns() as $columnName => $columnConfiguration) {
-            if (in_array($columnName, $existingColumnNames, true)) {
-                continue;
+            if (!in_array($columnName, $existingFieldNames, true)) {
+                $newFields[] = $columnName;
             }
+        }
+
+        if (empty($newFields)) {
+            return; // Nothing to add
+        }
+
+        // Step 1: Placeholder logic
+        $placeholder = '###FIELDS_GENERAL###';
+        if (str_contains($showItemsString, $placeholder)) {
+            $showItems->value = str_replace($placeholder, implode(', ', $newFields) . ',', $showItemsString);
+            return;
+        }
+
+        // Step 2: Split by --div-- and insert into first tab
+        $tabs = preg_split('/(\s*--div--;[^,]+,\s*)/', $showItemsString, -1, PREG_SPLIT_DELIM_CAPTURE);
 
-            $existingColumnNames[] = $columnName;
+        $inserted = false;
+        for ($i = 0; $i < count($tabs); $i++) {
+            if (!$inserted && str_starts_with(trim($tabs[$i]), '--div--')   && isset($tabs[$i + 1])) {
+                $existingFields = GeneralUtility::trimExplode(',', $tabs[$i + 1], true);
+
+                // Add only new fields
+                foreach ($newFields as $field) {
+                    if (!in_array($field, $existingFields, true)) {
+                        $existingFields[] = $field;
+                    }
+                }
+
+                $tabs[$i + 1] = implode(', ', $existingFields) . ',';
+                $inserted = true;
+            }
         }
 
-        $showItems->value = implode(', ', $existingColumnNames);
+        if ($inserted) {
+            $showItems->value = implode('', $tabs);
+        } else {
+            // No tabs at all: fallback to flat list
+            $existingFields = GeneralUtility::trimExplode(',', $showItemsString, true);
+
+            foreach ($newFields as $field) {
+                if (!in_array($field, $existingFields, true)) {
+                    $existingFields[] = $field;
+                }
+            }
+
+            $showItems->value = implode(', ', $existingFields);
```

Comment:
In row 121 $showItems is already "->value". So, now you call "->value->value". No tested, but that works?!